src/features/smartSearch/components/filters/EmailClick/index.tsx

```diff
@@ -220,7 +222,17 @@ const EmailClick = ({
                     getOptionDisabled={(link) =>
                       filter.config.links?.includes(link.id) || false
                     }
-                    noOptionsText={<Msg id={messageIds.misc.noOptionsLinks} />}
+                    noOptionsText={
+                      selectedEmail ? (
+                        selectedEmail.processed ? (
+                          <Msg id={messageIds.misc.noOptionsLinks} />
+                        ) : (
+                          <Msg id={messageIds.misc.noOptionsEmailNotSent} />
+                        )
+                      ) : (
+                        <Msg id={messageIds.misc.noOptionsInvalidEmail} />
+                      )
+                    }
```

Comment:
Hmmm... I'm not the biggest fan of this, since these checks are only relevant in the place that it is currently being performed, and I think the improvement in readability is offset by moving the logic away from these lines. But I can make a commit for this.

Reply:
I feel like Ziggi about nested ternaries, but I also don't love having mutable variables to solve it. I think the best solution is to create a function that can return in different conditional branches. But in this case, I think this is small enough to accept either.